rust/connlib/tunnel/src/client.rs

```diff
@@ -1050,6 +1050,9 @@ impl ClientState {
                 // We don't really expect these to change but let's update them anyway.
                 existing.ip.v4 = config.ipv4;
                 existing.ip.v6 = config.ipv6;
+
+                // These are safe to always update
+                existing.search_domain = config.search_domain.clone();
```

Comment:
I am not sure if this is enough, we probably also need to check in `maybe_update_tun_config` to actually push it to the host app? We should test this as well in the proptests.

Reply:
You were right - `update_dns_mapping` is a no-op if the upstream resolvers don't change.

I tested manually extensively in #8452 which, although ugly, works robustly.

Reply:
It was very painful working through a few failed refactors to get to that PR 🙃. Lots of complexity I see what you mean.